parser_library/src/workspaces/processor_file_impl.cpp

```diff
@@ -140,17 +140,57 @@ const std::set<std::string>& processor_file_impl::dependencies() { return depend
 
 const semantics::lines_info& processor_file_impl::get_hl_info()
 {
-    return last_analyzer_->source_processor().semantic_tokens();
+    if (last_analyzer_)
+        return last_analyzer_->source_processor().semantic_tokens();
+
+    const static semantics::lines_info empty_lines;
+    return empty_lines;
 }
 
+namespace {
+class empty_feature_provider final : public lsp::feature_provider
+{
+    // Inherited via feature_provider
+    virtual location definition(const std::string& document_uri, position pos) const override
+    {
+        return location(pos, document_uri);
+    }
+    virtual location_list references(const std::string& document_uri, position pos) const override
+    {
+        return location_list();
+    }
+    virtual lsp::hover_result hover(const std::string& document_uri, position pos) const override
+    {
+        return lsp::hover_result();
+    }
+    virtual lsp::completion_list_s completion(const std::string& document_uri,
+        position pos,
+        char trigger_char,
+        completion_trigger_kind trigger_kind) const override
+    {
+        return {};
+    }
```

Comment:
drop the virtual

Reply:
also I wonder whether we should have the capability of returning "we don't know yet".